com.unity.netcode.gameobjects/Components/HalfVector3.cs

```diff
@@ -0,0 +1,114 @@
+        /// </summary>
+        public ushort Z => Axis[2];
+
+        public ushort[] Axis;
+
+        private HalfVector3AxisToSynchronize m_HalfVector3AxisToSynchronize;
+
+        /// <summary>
+        /// The serialization implementation of <see cref="INetworkSerializable"/>
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <param name="serializer"></param>
+        public void NetworkSerialize<T>(BufferSerializer<T> serializer) where T : IReaderWriter
+        {
+            for (int i = 0; i < 3; i++)
+            {
+                if (m_HalfVector3AxisToSynchronize.SyncAxis[i])
+                {
+                    var axisValue = Axis[i];
+                    serializer.SerializeValue(ref axisValue);
+                    if (serializer.IsReader)
+                    {
+                        Axis[i] = axisValue;
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Converts the <see cref="HalfVector3"/> to a full precision <see cref="Vector3"/>
+        /// </summary>
+        /// <param name="vector3">the <see cref="Vector3"/> to store the full precision values in</param>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public void ToVector3(ref Vector3 vector3)
+        {
+            for (int i = 0; i < 3; i++)
+            {
+                if (m_HalfVector3AxisToSynchronize.SyncAxis[i])
+                {
+                    vector3[i] = Mathf.HalfToFloat(Axis[i]);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Converts a <see cref="Vector3"/> full precision to a <see cref="HalfVector3"/>
+        /// </summary>
+        /// <param name="vector3">the <see cref="Vector3"/> to convert to a half precision <see cref="Vector3"/></param>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public void FromVector3(ref Vector3 vector3)
+        {
+            for (int i = 0; i < 3; i++)
+            {
+                if (m_HalfVector3AxisToSynchronize.SyncAxis[i])
+                {
+                    Axis[i] = Mathf.FloatToHalf(vector3[i]);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Constructor that initializes the HalfVector3 along with its precision adjustment
+        /// </summary>
+        /// <remarks>
+        /// Note about decimal precision:
+        /// If you know that all components (x, y, and z) of a Vector3 will not exceed a specific threshold, then you
+        /// can increase the decimal precision of a Vector3 by increasing the decimalPrecision value. The decimal precision
+        /// valid values range from 0 to 4 decimal places to adjust up and back down (1000 to 1 : 1 to 0.001).
+        /// </remarks>
+        /// <param name="vector3">the vector3 to initialize the HalfVector3 with</param>
+        public HalfVector3(Vector3 vector3, HalfVector3AxisToSynchronize halfVector3AxisToSynchronize)
```

Comment:
Could we have another pair of constructors with the second parameter defaulting to (true, true, true)?

```public HalfVector3(Vector3 vector3) : this(vector3, new HalfVector3AxisToSynchronize())```

`HalfVector3AxisToSynchronize` is very descriptive but also a lot to type if users are using the default values anyway.

Reply:
I think if I were to rename that AxisToSynchronize it would help with the name, but it would be easy to have a default of all axis too.  👍